File: src/test/scala/tlschannel/helpers/NullSslContext.scala
package tlschannel.helpers

import javax.net.ssl.SSLContext
import javax.net.ssl.SSLContextSpi
import javax.net.ssl.KeyManager
import javax.net.ssl.TrustManager
import java.security.SecureRandom

class NullSslContextSpi extends SSLContextSpi {
  override def engineCreateSSLEngine() = new NullSslEngine
  def engineCreateSSLEngine(x$1: String, x$2: Int) = throw new UnsupportedOperationException
  def engineGetClientSessionContext() = throw new UnsupportedOperationException
  def engineGetServerSessionContext() = throw new UnsupportedOperationException
  def engineGetServerSocketFactory() = throw new UnsupportedOperationException
  def engineGetSocketFactory() = throw new UnsupportedOperationException
  def engineInit(x$1: Array[KeyManager], x$2: Array[TrustManager], x$3: SecureRandom): Unit =
    throw new UnsupportedOperationException
}

class NullSslContext extends SSLContext(new NullSslContextSpi, null, null)

File: src/test/scala/tlschannel/helpers/NullSslContext.java
package tlschannel.helpers;

import javax.net.ssl.*;
import java.security.SecureRandom;


public class NullSslContext extends SSLContext {
  public NullSslContext() {
    super(new NullSslContextSpi(), null, null);
  }

}

class NullSslContextSpi extends SSLContextSpi {

  @Override
  public SSLEngine engineCreateSSLEngine() {
    return new NullSslEngine();
  }

  @Override
  public SSLEngine engineCreateSSLEngine(String s, int i) {
    throw new UnsupportedOperationException();
  }

  @Override
  public SSLSessionContext engineGetClientSessionContext() {
    throw new UnsupportedOperationException();
  }

  @Override
  public SSLSessionContext engineGetServerSessionContext() {
    throw new UnsupportedOperationException();
  }
  @Override
  public SSLServerSocketFactory engineGetServerSocketFactory() {
    throw new UnsupportedOperationException();
  }

  @Override
  public SSLSocketFactory engineGetSocketFactory() {
    throw new UnsupportedOperationException();
  }

  @Override
  public void engineInit(KeyManager[] km, TrustManager[] tm, SecureRandom sc) {
    throw new UnsupportedOperationException();
  }

}